integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/validation/UUIDValidator.java

```diff
@@ -0,0 +1,29 @@
+package no.difi.meldingsutveksling.validation;
+
+import lombok.extern.slf4j.Slf4j;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+@Slf4j
+public class UUIDValidator implements ConstraintValidator<UUID, String> {
+
+    @Override
+    public void initialize(UUID constraint) {
+        // NOOP
+    }
+
+    public boolean isValid(String s, ConstraintValidatorContext context) {
+        return s == null || isUUID(s);
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    private boolean isUUID(String s) {
+        try {
+            java.util.UUID.fromString(s);
+            return true;
+        } catch (IllegalArgumentException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)